test/Amazon.Lambda.Tools.Integ.Tests/DeployProjectTests.cs

```diff
@@ -278,5 +278,184 @@ public async Task PackageFunctionAsLocalImageThenDeployWithDifferentECRTag()
+            command.FunctionName = functionName;
+            command.Role = await TestHelper.GetTestRoleArnAsync();
+            command.Runtime = "dotnet6";
+            command.EphemeralStorageSize = 750;
+            command.EnvironmentVariables = new System.Collections.Generic.Dictionary<string, string> { { "Key1", "Value1" } };
+            command.DisableInteractive = true;
+            var created = await command.ExecuteAsync();
+
+            try
+            {
+                Assert.True(created);
+                var getConfigResponse = await command.LambdaClient.GetFunctionConfigurationAsync(functionName);
+                Assert.Equal(750, getConfigResponse.EphemeralStorage.Size);
+                Assert.Single(getConfigResponse.Environment.Variables);
+
+                command = new DeployFunctionCommand(new TestToolLogger(_testOutputHelper), fullPath, new string[0]);
+                command.FunctionName = functionName;
+                command.Role = await TestHelper.GetTestRoleArnAsync();
+                command.Runtime = "dotnet6";
+                command.EphemeralStorageSize = 800;
+                command.EnvironmentVariables = new System.Collections.Generic.Dictionary<string, string> ();
+                command.DisableInteractive = true;
+
+                created = await command.ExecuteAsync();
+                Assert.True(created);
+
+                getConfigResponse = await command.LambdaClient.GetFunctionConfigurationAsync(functionName);
+                Assert.Equal(800, getConfigResponse.EphemeralStorage.Size);
+                Assert.Null(getConfigResponse.Environment);
+            }
+            finally
+            {
+                try
+                {
+                    await command.LambdaClient.DeleteFunctionAsync(functionName);
+                }
+                catch { }
+            }
+        }
+
+        [Fact]
+        public async Task DeployWithFunctionUrl()
+        {
+            using var httpClient = new HttpClient();
+            var assembly = this.GetType().GetTypeInfo().Assembly;
+
+            var toolLogger = new TestToolLogger(_testOutputHelper);
+
+            var fullPath = Path.GetFullPath(Path.GetDirectoryName(assembly.Location) + "../../../../../../testapps/TestHttpFunction");
+
+            var functionName = "DeployWithFunctionUrl-" + DateTime.Now.Ticks;
+
+            // Initial deployment with Function Url enabled and Auth will default to NONE.
+            var command = new DeployFunctionCommand(new TestToolLogger(_testOutputHelper), fullPath, new string[0]);
+            command.FunctionName = functionName;
+            command.Role = await TestHelper.GetTestRoleArnAsync();
+            command.Runtime = "dotnet6";
+            command.FunctionUrlEnable = true;
+            command.DisableInteractive = true;
+
+
+            var created = await command.ExecuteAsync();
+            try
+            {
```

Comment:
I get that the integration test does a couple of things and then has a final cleanup. Would it be possible to move each of those "couple of things" to separate, smaller private methods, so that it is easier to inspect what is being exercised and validated with each one?

Reply:
Cleanup up the integ test and added some test utility methods.

Reply:
Looks great!